### PR TITLE
Scan tuning refactor hub

### PR DESCRIPTION
**Pull request by gonidelis**
Needed for subsequent policies

### CODE REVIEW

**Comment by gonidelis** [comment]
Converting to draft until design is discussed with the team.

File: cub/cub/device/dispatch/tuning/tuning_scan.cuh [hunk 1]
```diff
@@ -293,7 +299,7 @@ struct policy_hub
 } // namespace detail
 
 // TODO(bgruber): deprecate this at some point when we have a better way to allow users to supply tunings
-template <typename AccumT, typename ScanOpT = ::cuda::std::plus<>>
-using DeviceScanPolicy = detail::scan::policy_hub<AccumT, ScanOpT>;
+template <typename AccumT, typename OffsetT, typename ScanOpT = ::cuda::std::plus<>>
+using DeviceScanPolicy = detail::scan::policy_hub<AccumT, OffsetT, ScanOpT>;
```
Comment (bernhardmgruber) [inline cub/cub/device/dispatch/tuning/tuning_scan.cuh:303]: Critical: This is a breaking change. We can work around this by supplying a dummy offset type that won't affect old tuning policies:
```suggestion
template <typename AccumT, typename ScanOpT = ::cuda::std::plus<>>
using DeviceScanPolicy = detail::scan::policy_hub<AccumT, int, ScanOpT>;
```

File: cub/benchmarks/bench/scan/exclusive/base.cuh [hunk 2]
```diff
@@ -30,6 +30,7 @@
 #include <cuda/std/__functional/invoke.h>
 
 #include <look_back_helper.cuh>
+#include <nvbench_helper.cuh>
```
Comment (bernhardmgruber) [inline cub/benchmarks/bench/scan/exclusive/base.cuh:33]: This change seems unreleated to this PR. Why is it needed?

**Review by bernhardmgruber — CHANGES_REQUESTED** [review]

File: cub/benchmarks/bench/scan/exclusive/base.cuh (hunk 2 above)
Comment (bernhardmgruber) [reply cub/benchmarks/bench/scan/exclusive/base.cuh:33]: Just found it here: #3157. Please rebase this PR then.

**Pull request closed without merging**